packages/grafana-e2e/src/flows/addPanel.ts

```diff
@@ -89,6 +81,14 @@ export const addPanel = (config?: Partial<AddPanelConfig>): any =>
     return e2e().wrap({ config: fullConfig });
   });
 
+// @todo this actually returns type `Cypress.Chainable`
+const closeOptions = (): any =>
```

Comment:
can these be async functions?

Reply:
Nope, and it's something that I dislike about Cypress. Its API is Promise-like, but not true promises. The asynchronous behaviour is abstracted away from us into a queue. It produces some ugly code that's difficult to debug.

Reply:
I agree with @stevenvachon this is a source of headaches, confusion and has made me rethink our choice of Cypress several times. We used puppeteer before Cypress and the code you wrote for puppeteer was more similar to the code you write on a daily base. Cypress has some pleasant features that puppeteer hasn't like the `dev` mode with time travel.

If this continues to confuse us and slow us down, maybe we should make this super clear in our e2e docs?

Reply:
I'll be putting together some documentation and will surely link to the Cypress page explaining their confusing API.

Reply:
Thank you @stevenvachon